### PR TITLE
[GitRest] Allowing customization of git directory through settings

### PR DESCRIPTION
**Pull request by hedasilv**
Allowing for the directory used by the Git RepositoryManagers to be customizable in 2 ways:
* Allowing to either use or ignore the `owner` parameter. That's because `owner` currently never changes. `name` currently maps to the tenantId, while `owner` is hardcoded. Therefore, the Git directory can be simplified by omitting `owner`.
* Allowing to not use a `baseDir` for the Git repo path. `baseDir` makes sense when we are mapping a volume in Docker, for example, where we will store Git data. But with `isomorphic-git` and custom filesystem implementations, that `baseDir` is not needed.

### CODE REVIEW

**Review by znewton — APPROVED** [review]

File: server/gitrest/packages/gitrest-base/src/utils/helpers.ts [hunk 1]
```diff
@@ -104,23 +104,20 @@ export async function retrieveLatestFullSummaryFromStorage(
...
+export function getRepoPath(name: string, owner?: string): string {
+    // `name` needs to be always present and valid.
+    if (!name || name === "" || path.parse(name).dir !== "") {
+        throw new NetworkError(400, `Invalid repo name provided.`);
     }
 
-    if (!name || name === "") {
-        throw new NetworkError(400, `Invalid arguments. A repo name must be provided.`);
+    // When `owner` is present, it needs to be valid.
+    if (owner && (owner === "" || path.parse(owner).dir !== "")) {
```
Comment (znewton) [inline server/gitrest/packages/gitrest-base/src/utils/helpers.ts:114]: nit: `owner` can't be truthy and also `=== ""`

File: server/gitrest/packages/gitrest-base/src/utils/helpers.ts [hunk 2]
```diff
@@ -104,23 +104,20 @@ export async function retrieveLatestFullSummaryFromStorage(
 /**
  * Retrieves the full repository path. Or throws an error if not valid.
  */
-export function getRepoPath(owner: string, name: string) {
-    if (!owner || owner === "") {
-        throw new NetworkError(400, `Invalid arguments. A repo owner must be provided.`);
+export function getRepoPath(name: string, owner?: string): string {
+    // `name` needs to be always present and valid.
+    if (!name || name === "" || path.parse(name).dir !== "") {
```
Comment (znewton) [inline server/gitrest/packages/gitrest-base/src/utils/helpers.ts:109]: nit: `!name === false` implies `name !== ""`

**Review by znewton — COMMENTED** [review]
Line 35 of gitrest/src/test/treebuilder.spec.ts needs to be changed to account for storageDir being an object

**Review by yangg-msft — APPROVED** [review]
:shipit:

**Pull request merged**